pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotRealtimeTableResource.java

```diff
@@ -103,12 +104,14 @@ public class PinotRealtimeTableResource {
   @ApiOperation(value = "Pause consumption of a realtime table", notes = "Pause the consumption of a realtime table")
   public Response pauseConsumption(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "Comment on pausing the consumption") @QueryParam("comment") String comment,
```

Comment:
The new argument is optional, right? 

I suppose the compatibility story is that we will keep it optional for one release, and then turn it into a mandatory argument?

Reply:
Also, how about we introduce default comment string in this class, and pass that down?

Reply:
Yes its optional. I don’t think we need to make it mandatory in future.
 
> Also, how about we introduce default comment string in this class, and pass that down?

Yes that's a good point. I am thinking of assigning the default value at the caller code so that right default context is set.

Reply:
I don't quite understand "caller code", but I think we should populate it with the default at the earliest point upon entry into the pinot-controller. I hope you don't mean pinot client or UI because not everyone uses it

Reply:
Also, why not make it mandatory eventually? It is always a good practice for operators to indicate reason

Reply:
> I don't quite understand "caller code"

on second thought, the other places where table might be paused will most likely be system flows (e.g. storage quota breach) so there we don’t need a default hence having default comment at the class itself makes sense.

Reply:
> Also, why not make it mandatory eventually? It is always a good practice for operators to indicate reason

No strong reason just avoiding the restriction as the comment input is not a crucial input for pinot.

Reply:
Let us make it mandatory after one release then. In any reasonable production instance, it is always best practice to leave comments when service is affected due to manual action. The comment can then be imported into alerting or other mechanisms that admins may build over Pinot.

please create an issue, keep track of it, and fix it when the next release is completed.

thanks

Reply:
Got it. Will create an issue right after the changes are in.